src/adapters/websocket-adapter.ts

```diff
@@ -0,0 +1,79 @@
+
+import {io , Socket} from 'socket.io-client'
+import { getCookie, cookieName } from '../util/Authentication';
+const socketURL = window.location.hostname === 'localhost'? 'ws://localhost:8000': 'wss://shelf-share.onrender.com'
+
+const jwtCookie = getCookie(cookieName)as { token: string }
+const jwtToken = jwtCookie ? jwtCookie.token: null;
```

Comment:
Can you get the cookie inside of the adapters? That way we always fetch the most up to date value.